app/src/main.py

```
@@ -575,33 +670,130 @@ def expose_metrics(self):
+            now = datetime.now(timezone.utc)
+            held = self._held_symbols()
+            scheduled = self._scheduled_symbols()
+            # Add new + revive missing in one pass: any held symbol without a
+            # live job fires immediately.
+            for symbol in held - scheduled:
+                self._arm_symbol(symbol, 0, now)
+            # Remove departed symbols' idle jobs (belt-and-braces with the
+            # in-flight membership re-check in _scrape_symbol).
+            for symbol in scheduled - held:
+                self.scheduler.remove_job(_scrape_job_id(symbol))
+        except Exception as e:
```

Ruff (0.15.21)

[warning] Do not catch blind exception: `Exception`

(BLE001)